### PR TITLE
Accomodate latest miners updates

### PR DESCRIPTION
**Pull request by LuKePicci**
Finally cleaned up the legacy baggage. 
I quickly updated pm management in 0cpuminer, if you spend some time in porting cpuOPT and test it as pm let me know about issues.
Some updates from @delawr0190's #244 are included here.

### CODE REVIEW

**Comment by LuKePicci** [comment]
@delawr0190 if you plan to test latest changes use this branch as codebase.

**Comment by delawr0190** [comment]
I’ll retest it all tonight.

File: foreman/miner-watch [hunk 1]
```diff
@@ -30,93 +30,16 @@ check_miner() {
...
-                API_PORT=$(get_ccminer_port)
-                ;;
-            "spccminer")
-                MINER="ccminer-sp-hash"
-                API_PORT=$(get_ccminer_port)
-                ;;
-            "suprminer")
-                API_PORT=$(get_ccminer_port)
-                ;;
-            "tpccminer")
```
Comment (delawr0190) [inline foreman/miner-watch:104]: I don't think this miner has a pm file yet, so this is still needed.

Comment (papampi) [reply foreman/miner-watch:104]: Added TPccminer pm, please check,  https://github.com/papampi/nvOC_miners/pull/30/commits/6ee6172b7eeeb2c3c0c56cac6b6949636c57ecfe

**Review by papampi — APPROVED** [review]

**Comment by papampi** [comment]
Is there any commits from 3.1-dev-prep need to be merged to 3.1-dev?

**Comment by LuKePicci** [comment]
Nope, I manually applied all changes here already.

**Comment by papampi** [comment]
So it can be removed?

**Comment by LuKePicci** [comment]
I think @delawr0190 is the owner of the branch, so maybe he still need it, ask to him.

**Pull request merged**